src/Entity/User.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+// Totalement inutile car remplacé par User2.php
+
+namespace App\Entity;
+
+use App\Repository\UserRepository;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
+
+#[ORM\Entity(repositoryClass: UserRepository::class)]
+#[UniqueEntity('email', message: 'Cet e-mail est déjà utilisé.')]
+class User
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\Column(length: 255)]// prénom
+    #[Assert\Length(max: 255)]
+    private ?string $firstName = null;
+
+    #[ORM\Column(length: 255)] // nom de famille
+    #[Assert\Length(max: 255)]
+    private ?string $lastName = null;
+
+    #[ORM\Column(length: 255, unique: true)]
+    #[Assert\NotBlank]
+    #[Assert\Email]
+    #[Assert\Length(max: 255)]
+    private ?string $email = null;
```

Comment:
Item si l'email ne peut pas être null à cause du #[Assert\NotBlank] alors
```suggestion
    private string $email = "";
```